sources/rest_api/auth.py

```diff
@@ -1,258 +1,37 @@
-from base64 import b64encode
-import math
 from typing import (
-    Dict,
     Final,
     Literal,
-    Optional,
-    Union,
-    Any,
-    cast,
-    Iterable,
-    TYPE_CHECKING,
 )
-from dlt.sources.helpers import requests
-from requests.auth import AuthBase
-from requests import PreparedRequest  # noqa: I251
-import pendulum
 
-from dlt.common.exceptions import MissingDependencyException
-
-from dlt import config, secrets
-from dlt.common import logger
 from dlt.common.configuration.specs.base_configuration import configspec
-from dlt.common.configuration.specs import CredentialsConfiguration
-from dlt.common.configuration.specs.exceptions import NativeValueError
-from dlt.common.typing import TSecretStrValue
-
-if TYPE_CHECKING:
-    from cryptography.hazmat.primitives.asymmetric.types import PrivateKeyTypes
-else:
-    PrivateKeyTypes = Any
-
-TApiKeyLocation = Literal[
-    "header", "cookie", "query", "param"
-]  # Alias for scheme "in" field
-
 
-class AuthConfigBase(AuthBase, CredentialsConfiguration):
-    """Authenticator base which is both `requests` friendly AuthBase and dlt SPEC
-    configurable via env variables or toml files
-    """
-
-    pass
+from dlt.sources.helpers.rest_client import auth
 
 
 @configspec
-class BearerTokenAuth(AuthConfigBase):
+class BearerTokenAuth(auth.BearerTokenAuth):
     type: Final[Literal["http"]] = "http"  # noqa: A003
     scheme: Literal["bearer"] = "bearer"
-    token: TSecretStrValue = secrets.value
-
-    def __init__(self, token: TSecretStrValue = secrets.value) -> None:
-        self.token = token
-
-    def parse_native_representation(self, value: Any) -> None:
-        if isinstance(value, str):
-            self.token = cast(TSecretStrValue, value)
-        else:
-            raise NativeValueError(
-                type(self),
-                value,
-                f"BearerTokenAuth token must be a string, got {type(value)}",
-            )
-
-    def __call__(self, request: PreparedRequest) -> PreparedRequest:
-        request.headers["Authorization"] = f"Bearer {self.token}"
-        return request
 
 
 @configspec
-class APIKeyAuth(AuthConfigBase):
+class APIKeyAuth(auth.APIKeyAuth):
```

Comment:
@rudolfix is this a proper way to extend the configspec with an additional `type` field?

Reply:
yes! but why are you subclassing? we just import the Auth implementations from the core... we got rid of the openAPI literals below and we should live with this :)

tldr;> just import core Auth into the `auth` module here

Reply:
> yes! but why are you subclassing? we just import the Auth implementations from the core... we got rid of the openAPI literals below and we should live with this :)
I see, I thought we wanted to keep the OpenAPI fields in the verified source versions. Will update.